src/mkobj.c

```diff
@@ -761,7 +761,7 @@ static const char dknowns[] = { WAND_CLASS,   RING_CLASS, POTION_CLASS,
 void
 clear_dknown(struct obj *obj)
 {
-    obj->dknown = strchr(dknowns, obj->oclass) ? 0 : 1;
+    obj->dknown = index(dknowns, obj->oclass) ? 0 : 1;
```

Comment:
Did you mean to revert this one part of 99a93fe50?

Reply:
That must have got in by mistake.